packages/babel-template/src/parse.js

```diff
@@ -103,13 +103,17 @@ function placeholderVisitorHandler(
   state.placeholders.push({
     name,
     type,
-    resolve: ast => resolveAncestors(ast, ancestors),
+    // $FlowFixMe
+    resolve: (ast: BabelNode) => resolveAncestors(ast, ancestors),
     isDuplicate: state.placeholderNames.has(name),
   });
   state.placeholderNames.add(name);
 }
 
-function resolveAncestors(ast: BabelNodeFile, ancestors: TraversalAncestors) {
+function resolveAncestors<T: BabelNode>(
+  ast: T,
```

Comment:
I'm not sure these types really make sense. The ancestors here could be any type. There's not really a way for the type checker to ever know what parent types a node could be inside of. Same for `placeholderVisitorHandler` above.

Reply:
How would you change them? Should I change `TraversalAncestors<T>` to `TraversalAncestors<*>`

Reply:
I think this means that the parent can be any `BabelNode`, as opposed to literally `any`.

Reply:
After looking at it again, it really makes no sense. So I reverted that part in the latest commit.